components/raftstore/src/store/peer.rs

```diff
@@ -2278,7 +2279,13 @@ where
                 return false;
             }
             Ok(RequestPolicy::ReadIndex) => return self.read_index(ctx, req, err_resp, cb),
-            Ok(RequestPolicy::ProposeNormal) => self.propose_normal(ctx, req),
+            Ok(RequestPolicy::ProposeNormal) => {
+                if disk::WRITE_PERMISSION.load(Ordering::Acquire) {
```

Comment:
Better to allow configuration change on leader.

Reply:
And you need to write function
```rust
fn disk_almost_full() -> bool
```
then add failpoint in it.

Reply:
done

Reply:
> Better to allow configuration change on leader.

yes, Configure change allowed when leader disk full.